migration/manager.py

```diff
@@ -22,18 +21,19 @@ def get_tools_installed_in_account(self) -> list[ExternalTool]:
         tools = [ExternalTool(id=tool_dict['id'], name=tool_dict['name']) for tool_dict in results]
         return tools
 
-    def get_courses_in_terms(self, term_ids: list[int], bail_after: int | None = None) -> list[Course]:
-        limit_per_term = None
-        if bail_after is not None:
-            limit_per_term = bail_after // len(term_ids)
+    def get_courses_in_terms(self, term_ids: list[int], limit: int | None = None) -> list[Course]:
+        limit_chunks = None
+        if limit is not None:
+            limit_chunks = chunk_integer(limit, len(term_ids))
```

Comment:
I added a utility function and logic so that `limit` could be used for this method with multiple terms. `chunk_integer` ensures that -- except for some edge cases -- the result should have courses from each term. If `limit` isn't specified, there should not be any effect, all results will be returned.